### ISSUE TITLE
Split mp3 according to its chapters

### ISSUE THREAD

**Issue by CrustyApplesniffer** [openaudible/openaudible#92]
> That's unfortunate. I've searched online for a good open source splitter. You either have to pay for them, or split them all manually.
> 
> _Originally posted by @Gagtech in https://github.com/openaudible/openaudible/issues/43#issuecomment-431226396_

I've just created a [new repository](https://github.com/PrometeoConseil/AudioBookSplitter)  of a command line tool that extracts the chapter list and splits a source mp3 audiobook into several ones.

**Comment by openaudible** [comment]
I like one file per book, but for those who like split files, this looks good. Thanks for posting.

**Comment by KaraokeAmerica** [comment]
> I've just created a [new repository](https://github.com/PrometeoConseil/AudioBookSplitter) of a command line tool that extracts the chapter list and splits a source mp3 audiobook into several ones.

I apologize for being so ignorant, but is there a "how to" document somewhere for using your splitting tool?

Thank you!

**Comment by guysoft** [comment]
@CrustyApplesniffer Hey, it has no release, and no instructions how to use.
I also like split chapters, but that repo is not really usable in that state.

**Comment by guysoft** [comment]
Hey,
I will also note that AAXtoMP3 also supported chapter splitting:
https://github.com/KrumpetPirate/AAXtoMP3/blob/master/AAXtoMP3#L427

At the moment I am not using OpenAudible because this feature is missing.

**Comment by KaraokeAmerica** [comment]
> I will also note that AAXtoMP3 also supported chapter splitting:
> https://github.com/KrumpetPirate/AAXtoMP3/blob/master/AAXtoMP3#L427

Thanks for this info. Since I am not a coder, do you know how I can use this in Windows10? I'm not afraid of a command line, but it seems that there is a lot of a priori knowledge involved when you use software/scripts from GitHub and I don't necessarily have that knowledge.

When I look at the script you linked to above it appears to have a lot of executables that it "needs" to do this work. They are not part of the repository I just downloaded.

I appreciate any direction you can give me here!

**Comment by guysoft** [comment]
@KaraokeAmerica You need to use Linux Subsystem, which is a windows tool to run linux stuff. Someone posted that it works here: https://github.com/KrumpetPirate/AAXtoMP3/issues/66

But its not a point-and-click walk in the park. You will need to know lots of small things which I am not really the one to guide you though because I am not a windows user.

**Comment by CrustyApplesniffer** [comment]
@KaraokeAmerica, I've added a short documentation and 2 installers (for 32bits and for 64bits Windows versions)

**Comment by CrustyApplesniffer** [comment]
@guysoft, I've fixed the lack of documentation, and added a first release.
Originally, I just wanted to do a Proof Of Concept to find the command lines for chapter extraction. I had not planned to make it a tool that others could use.

**Comment by KaraokeAmerica** [comment]
> @KaraokeAmerica, I've added a short documentation and 2 installers (for 32bits and for 64bits Windows versions)

I'm sorry, but I don't see those. Is it in the GitHub repository?

Also, thank you so much for the help!

**Comment by guysoft** [comment]
I'd appreciate an executable ``.exe``. Because I am using linux and although I can use mono to run .net stuff. I can't install .msi files.

**Comment by CrustyApplesniffer** [comment]
@guysoft : I've made a zip file with all needed exe and dll [Prometeo.Audiobook.splitter.zip](https://github.com/PrometeoConseil/AudioBookSplitter/releases/download/v1.1.81.7000/Prometeo.Audiobook.splitter.zip)
You can find it in the release assets https://github.com/PrometeoConseil/AudioBookSplitter/releases/tag/v1.1.81.7000

**Comment by CrustyApplesniffer** [comment]
@KaraokeAmerica : they are in the [release page](https://github.com/PrometeoConseil/AudioBookSplitter/releases/tag/v1.1.81.7000)

**Comment by KaraokeAmerica** [comment]
> I've made a zip file with all needed exe and dll [Prometeo.Audiobook.splitter.zip](https://github.com/PrometeoConseil/AudioBookSplitter/releases/download/v1.1.81.7000/Prometeo.Audiobook.splitter.zip)
> You can find it in the release assets https://github.com/PrometeoConseil/AudioBookSplitter/releases/tag/v1.1.81.7000

I am using the command line tool, but keep getting the following error:

```
System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.LegacyNormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.GetFullPathInternal(String path)
   at System.IO.Directory.InternalCreateDirectoryHelper(String path, Boolean checkHost)
   at AudioBookSplitterCmd.Program.Main(String[] args) in D:\Personnel\_repo\Audio Book Splitter\AudioBookSplitterCmd\Program.cs:line 90
Press any key to exit the program.
```

I have tried many variations, but the last one ran from a dir I created in the root of C to eliminate any possibility that the path actually is too long:

`C:\Prometeo Audiobook splitter>AudioBookSplitterCmd.exe -s "C:\Prometeo Audiobook splitter\SUBDIR\FILE_NAME.mp3" -o "C:\Prometeo Audiobook splitter\SUBDIR\output"`

Particularly strange is the last line in the error I'm getting as it looking for a path that contains drive "D" which is not called at all in my command line. When I look at some of the files like "CommandLine.xml" it shows path names all over the place that reference drives I don't even have. I admit I'm not a coder so I may be misinterpreting what they mean, but I wanted to give you my feedback.

I also tried the MSI after this all happened. It seemed to simply copy all the same files from the zip to a folder in my "Program Files".

Any thoughts?

**Comment by CrustyApplesniffer** [comment]
Did you try ```C:\Prometeo Audiobook splitter>AudioBookSplitterCmd.exe -s "C:\Prometeo Audiobook splitter\SUBDIR\FILE_NAME.mp3" -o "C:\output"```?

Don't worry about the D:\Personnel\... it is just the location where I store my project.
You can remove all pdf and xml files in the folder.

**Comment by guysoft** [comment]
@openaudible is there a way to run a "post download and convert script"?

Ok, finally got time to test this.

So what I did to get this working on linux was:
1. Download and unzop https://github.com/PrometeoConseil/AudioBookSplitter/releases/download/v1.1.81.7000/Prometeo.Audiobook.splitter.zip 

2. remove ``ffprobe.exe`` and ``ffmpeg.exe` then replace them with a symlink from the sytem:
```
ln -s /usr/bin/ffprobe ffprobe.exe
ln -s /usr/bin/ffmpeg ffmpeg.exe
```

3. Ran the cmd like this using mono:
```
mono AudioBookSplitterCmd.exe -s 'book.mp3'  -o .
```

Its working.
It might be nice also to extract the image cover.


As in after the download and convert has completed pass the MP3 to a script on the system? That we I could automate the creation and conversion of books (which takes time).
Or have some kind of plugin system.
